packages/dds/tree/src/test/tree/visit.spec.ts

```diff
@@ -184,4 +184,48 @@ describe("visit", () => {
         ];
         testTreeVisit(delta, expected);
     });
+
+    it("move children", () => {
+        const moveId: Delta.MoveId = brand(1);
+        const moveOut: Delta.MoveOut = {
+            type: Delta.MarkType.MoveOut,
+            count: 2,
+            moveId,
+        };
+
+        const moveIn: Delta.MoveIn = {
+            type: Delta.MarkType.MoveIn,
+            moveId,
+        };
+
+        const delta: Delta.Root = new Map([[
+            rootKey,
+            [{
+                type: Delta.MarkType.Modify,
+                fields: new Map([[
+                    fooKey,
+                    [2, moveOut, 3, moveIn],
+                ]]),
+            }],
+        ]]);
+
+        const expected: VisitScript = [
+            ["enterField", rootKey],
+            ["enterNode", 0],
+            ["enterField", fooKey],
+            ["onMoveOut", 2, 2, moveId],
+            ["exitField", fooKey],
+            ["exitNode", 0],
+            ["exitField", rootKey],
+            ["enterField", rootKey],
+            ["enterNode", 0],
+            ["enterField", fooKey],
+            ["onMoveIn", 3, 2, moveId],
```

Comment:
Shouldn't this be:
```suggestion
            ["onMoveIn", 5, 2, moveId],
```

Reply:
Your suggestion was the result before my change to visitDelta, but I believe we want the indices to reflect the state of the tree after the preceding move out, which removed two nodes from earlier in this trait.

Reply:
I believe the same, but somehow I'm still arriving at the 5.
I'm reading the changeset (`[2, moveOut, 3, moveIn]`) as "2 untouched nodes, 2 nodes being moved out, 3 untouched nodes, 2 nodes being moved in", so before any operations there are 7 nodes in the trait. During the second pass we therefore have 7-2 = 5 nodes remaining that need to be skipped over.

Reply:
You're right